documentation/bolt_installing_modules.md

```diff
@@ -329,44 +329,136 @@ module-install:
+   modules: []
+   ```
+
+1. Create a file named `Puppetfile` in your project directory and add the
+   modules you want to install, including each module's dependencies, to
+   the Puppetfile. For example:
+
+   ```ruby
+   # Modules from a private git repository
+   mod 'private-module', git: 'https://github.com/bolt-user/private-module.git', ref: 'main'
+
+   # Modules from an alternate Forge
+   mod 'puppetlabs/apache', '5.7.0'
+   mod 'puppetlabs/stdlib', '6.5.0'
+   mod 'puppetlabs/concat', '6.3.0'
+   mod 'puppetlabs/translate', '2.2.0'
+   ```
+
+1. Install the modules in the Puppetfile without resolving dependencies:
+
+   _\*nix shell command_
+
+   ```shell
+   bolt module install --no-resolve
+   ```
+
+   _PowerShell cmdlet_
+
+   ```powershell
+   Install-BoltModule -NoResolve
+   ```
+
+Bolt installs modules in the Puppetfile to the `.modules/` directory in your
+project directory. If you need to add or remove modules to the project, update
+your Puppetfile and run the above command again.
+
+## Migrate a project to use module management
+
+If you created a Bolt project before module dependency management was
+introduced, you can use Bolt's built-in migration tool to [update your
+project](projects.md#migrate-a-bolt-project).
+
+### Manual migration
+
+In some cases, Bolt is unable to resolve module dependencies and manage your
+project's modules for you. If Bolt can't resolve your module dependencies, you
+must manage your project's Puppetfile manually and use Bolt to install the
+modules listed in the Puppetfile without resolving dependencies. The most common
+scenario where Bolt can't resolve module dependencies is when a project includes
+git modules that are in a repository other than a public GitHub repository.
+
+The module management feature makes changes to configuration files and
+where modules are installed. To migrate your project, do the following:
+
+1. Move any configuration under `puppetfile` in your configuration files
+   under `module-install`. This includes both `bolt-defaults.yaml` and
+   `bolt-project.yaml` files. For example, the following file:
+
+   ```yaml
+   # bolt-project.yaml
+   name: myproject
+   puppetfile:
+     forge:
+       baseurl: https://myforge.example.com
+     proxy: https://myproxy.example.com:8080
+   ```
+
+   Becomes:
+
+   ```yaml
+   # bolt-project.yaml
+   name: myproject
+   module-install:
+     forge:
+       baseurl: https://myforge.example.com
+     proxy: https://myproxy.example.com:8080
+   ```
+
+1. Set the `modules` option in your `bolt-project.yaml` file to an
+   empty array:
+
+   ```yaml
+   # bolt-project.yaml
+   name: myproject
+   modules: []
+   ```
+
+1. Delete all modules in the `modules/` directory of your project.
+
+1. Copy all modules from the `site-modules/` and `site/` directories of your
+   project to the `modules/` directory. If you have configured a `modulepath` for
+   your project, you might need to move modules from different directories.
```

Comment:
The `modulepath` would still work, right? Why would a user need to move modules from directories they've specified into `modules/`?

Reply:
Ah, true. Changed that line to say you don't need to do this step if `modulepath` is configured.